tests/driver_sx1280/main.c

```diff
@@ -0,0 +1,366 @@
+#include "msg.h"
+#include "thread.h"
+#include "shell.h"
+
+#include "net/lora.h"
+#include "net/netdev.h"
+#include "net/netdev/lora.h"
+
+#include "sx1280.h"
+#include "sx1280_params.h"
+#include "sx1280_netdev.h"
+
+#define SX1280_MSG_QUEUE        (8U)
+#define SX1280_STACKSIZE        (THREAD_STACKSIZE_DEFAULT)
+#define SX1280_MSG_TYPE_ISR     (0x3456)
+#define SX1280_MAX_PAYLOAD_LEN  (128U)
+
+static char stack[SX1280_STACKSIZE];
+static kernel_pid_t _recv_pid;
+
+static char message[SX1280_MAX_PAYLOAD_LEN];
+
+static sx1280_t sx1280;
+
+static void _event_cb(netdev_t *dev, netdev_event_t event)
+{
+    if (event == NETDEV_EVENT_ISR) {
+        msg_t msg;
+        msg.type = SX1280_MSG_TYPE_ISR;
+        if (msg_send(&msg, _recv_pid) <= 0) {
+            puts("sx1280_netdev: possibly lost interrupt.");
+        }
+    }
+    else {
+        switch (event) {
+        case NETDEV_EVENT_RX_STARTED:
+            puts("Data reception started");
+            break;
+
+        case NETDEV_EVENT_RX_COMPLETE:
+        {
+            size_t len = dev->driver->recv(dev, NULL, 0, 0);
+            netdev_lora_rx_info_t packet_info;
+            dev->driver->recv(dev, message, len, &packet_info);
+            printf(
+                "Received: \"%s\" (%d bytes) - [RSSI: %i, SNR: %i]\n",
+                message, (int)len, packet_info.rssi, (int)packet_info.snr);
+            netopt_state_t state = NETOPT_STATE_RX;
+            dev->driver->set(dev, NETOPT_STATE, &state, sizeof(state));
+        }
+        break;
+
+        case NETDEV_EVENT_TX_COMPLETE:
+            puts("Transmission completed");
+            break;
+
+        case NETDEV_EVENT_TX_TIMEOUT:
+            puts("Transmission timeout");
+            break;
+
+        default:
+            printf("Unexpected netdev event received: %d\n", event);
+            break;
+        }
+    }
+}
+
+void *_recv_thread(void *arg)
+{
+    netdev_t *netdev = arg;
+
+    static msg_t _msg_queue[SX1280_MSG_QUEUE];
+
+    msg_init_queue(_msg_queue, SX1280_MSG_QUEUE);
+
+    while (1) {
+        msg_t msg;
+        msg_receive(&msg);
+        if (msg.type == SX1280_MSG_TYPE_ISR) {
+            netdev->driver->isr(netdev);
+        }
+        else {
+            puts("Unexpected msg type");
+        }
+    }
+}
+
+static void _get_usage(const char *cmd)
+{
+    printf("Usage: %s get <type|freq|bw|sf|cr|random>\n", cmd);
+}
+
+
+static void _usage_freq(void)
+{
+    printf("Usage: use freq between 2400000000 + (bw/2) and 2500000000 - (bw/2) (Hz) !\n");
+}
+
+static void _usage_bw(void)
+{
+    printf("Usage: use 200, 400, 800, 1600 (kHz)\n");
+
+}
+
+static void _usage_sf(void)
+{
+    printf("Usage: use SF between 5 and 12\n");
+}
+
+static void _usage_cr(void)
+{
+    printf(
+        "Usage: use\n \
+    RAL_LORA_CR_4_5 = 0\n \
+    RAL_LORA_CR_4_6 = 1\n \
+    RAL_LORA_CR_4_7 = 2\n \
+    RAL_LORA_CR_4_8 = 3\n \
+    RAL_LORA_CR_LI_4_5 = 4\n \
+    RAL_LORA_CR_LI_4_6 = 5\n \
+    RAL_LORA_CR_LI_4_8 = 6\n");
```

Comment:
```suggestion
    RAL_LORA_CR_4_5 = 0\n \
    RAL_LORA_CR_4_6 = 1\n \
    RAL_LORA_CR_4_7 = 2\n \
    RAL_LORA_CR_4_8 = 3\n \
    RAL_LORA_CR_LI_4_5 = 4\n \
    RAL_LORA_CR_LI_4_6 = 5\n \
    RAL_LORA_CR_LI_4_8 = 6\n");
```

Can you change to non `RAL` defines and use `LORA_CR_**` instead? Lets try to keep all user facing code non `RAL` specific, specially if you think we can remove that code.